// @dynamic-framework/ui/components/d-input-switch2.js
import { proxyCustomElement, HTMLElement, createEvent, h } from '@stencil/core/internal/client';

const DInputSwitch = /*@__PURE__*/ proxyCustomElement(class DInputSwitch extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
    this.eventChange = createEvent(this, "eventChange", 7);
    this.changeHandler = (event) => {
      const value = event.target.checked;
      this.internalIsChecked = value;
      this.eventChange.emit(value);
    };
    this.label = undefined;
    this.innerId = undefined;
    this.name = undefined;
    this.isChecked = undefined;
    this.isDisabled = undefined;
    this.isReadonly = undefined;
    this.internalIsChecked = undefined;
  }
  watchIsCheckedHandler(newValue) {
    this.internalIsChecked = newValue;
  }
  connectedCallback() {
    this.internalIsChecked = this.isChecked;
    if (this.isReadonly) {
      this.el.onclick = () => false;
    }
  }
  render() {
    return (h("div", { class: "form-check form-switch" }, h("input", { id: this.innerId, name: this.name, onChange: this.changeHandler, class: "form-check-input", type: "checkbox", role: "switch", checked: this.internalIsChecked, disabled: this.isDisabled }), !!this.label && (h("label", { class: "form-check-label", htmlFor: this.innerId }, this.label))));
  }
  get el() { return this; }
  static get watchers() { return {
    "isChecked": ["watchIsCheckedHandler"]
  }; }
}, [0, "d-input-switch", {
    "label": [1],
    "innerId": [1, "inner-id"],
    "name": [1],
    "isChecked": [4, "is-checked"],
    "isDisabled": [4, "is-disabled"],
    "isReadonly": [4, "is-readonly"],
    "internalIsChecked": [32]
  }]);
function defineCustomElement() {
  if (typeof customElements === "undefined") {
    return;
  }
  const components = ["d-input-switch"];
  components.forEach(tagName => { switch (tagName) {
    case "d-input-switch":
      if (!customElements.get(tagName)) {
        customElements.define(tagName, DInputSwitch);
      }
      break;
  } });
}

export { DInputSwitch as D, defineCustomElement as d };
