// @dynamic-framework/ui/components/d-input-check2.js
import { proxyCustomElement, HTMLElement, createEvent, h } from '@stencil/core/internal/client';

const DInputCheck = /*@__PURE__*/ proxyCustomElement(class DInputCheck extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
    this.eventChange = createEvent(this, "eventChange", 7);
    this.changeHandler = (event) => {
      const { checked, value } = event.target;
      this.eventChange.emit({
        isChecked: checked,
        value,
      });
    };
    this.type = undefined;
    this.name = undefined;
    this.label = undefined;
    this.isChecked = false;
    this.innerId = undefined;
    this.isDisabled = false;
    this.isIndeterminate = undefined;
    this.value = undefined;
  }
  render() {
    if (!this.label) {
      return (h("input", { onChange: this.changeHandler, class: "form-check-input", type: this.type, name: this.name, id: this.innerId, value: this.value, checked: this.isChecked, disabled: this.isDisabled, indeterminate: this.isIndeterminate }));
    }
    return (h("div", { class: "form-check" }, h("input", { onChange: this.changeHandler, class: "form-check-input", type: this.type, name: this.name, id: this.innerId, value: this.value, checked: this.isChecked, disabled: this.isDisabled, indeterminate: this.isIndeterminate }), h("label", { class: "form-check-label", htmlFor: this.innerId }, this.label)));
  }
}, [0, "d-input-check", {
    "type": [1],
    "name": [1],
    "label": [1],
    "isChecked": [4, "is-checked"],
    "innerId": [1, "inner-id"],
    "isDisabled": [4, "is-disabled"],
    "isIndeterminate": [4, "is-indeterminate"],
    "value": [1]
  }]);
function defineCustomElement() {
  if (typeof customElements === "undefined") {
    return;
  }
  const components = ["d-input-check"];
  components.forEach(tagName => { switch (tagName) {
    case "d-input-check":
      if (!customElements.get(tagName)) {
        customElements.define(tagName, DInputCheck);
      }
      break;
  } });
}

export { DInputCheck as D, defineCustomElement as d };
