// @dynamic-framework/ui/components/d-modal.js
import { proxyCustomElement, HTMLElement, createEvent, h } from '@stencil/core/internal/client';
import { P as PREFIX_BS } from './component-config.js';
import { d as defineCustomElement$2 } from './d-icon2.js';

const DModal$1 = /*@__PURE__*/ proxyCustomElement(class DModal extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
    this.eventClose = createEvent(this, "eventClose", 7);
    this.closeHandler = () => {
      this.eventClose.emit();
    };
    this.name = undefined;
    this.isStatic = undefined;
    this.isScrollable = undefined;
    this.isCentered = undefined;
    this.isFullScreen = undefined;
    this.fullScreenFrom = undefined;
    this.modalSize = undefined;
    this.showCloseButton = undefined;
    this.footerActionPlacement = 'fill';
  }
  componentWillLoad() {
    this.header = !!this.el.querySelector('[slot="header"]');
    this.body = !!this.el.querySelector('[slot="body"]');
    this.footer = !!this.el.querySelector('[slot="footer"]');
  }
  fullScreenClass() {
    if (this.isFullScreen) {
      if (this.fullScreenFrom) {
        return `modal-fullscreen-${this.fullScreenFrom}-down`;
      }
      return 'modal-fullscreen';
    }
    return '';
  }
  generateModalDialogClasses() {
    return {
      'modal-dialog': true,
      'modal-dialog-centered': !!this.isCentered,
      'modal-dialog-scrollable': !!this.isScrollable,
      [`modal-${this.modalSize}`]: !!this.modalSize,
      [this.fullScreenClass()]: !!this.isFullScreen,
    };
  }
  render() {
    return (h("div", Object.assign({ class: "modal fade show", id: this.name, tabindex: "-1", "aria-labelledby": `${this.name}Label`, "aria-hidden": "false" }, this.isStatic && ({
      [`data-${PREFIX_BS}backdrop`]: 'static',
      [`data-${PREFIX_BS}keyboard`]: 'false',
    })), h("div", { class: this.generateModalDialogClasses() }, h("div", { class: "modal-content" }, (this.header || this.showCloseButton) && (h("div", { class: "modal-header" }, this.showCloseButton && (h("button", { type: "button", class: "d-modal-close", "aria-label": "Close", onClick: this.closeHandler }, h("d-icon", { icon: "x-lg" }))), this.header && (h("div", { class: "d-modal-slot" }, h("slot", { name: "header" }))))), this.body && (h("div", { class: "d-modal-slot modal-body" }, h("slot", { name: "body" }))), this.footer && (h("div", { class: "d-modal-separator" })), this.footer && (h("div", { class: `d-modal-slot modal-footer d-modal-action-${this.footerActionPlacement}` }, h("slot", { name: "footer" })))))));
  }
  get el() { return this; }
}, [4, "d-modal", {
    "name": [1],
    "isStatic": [4, "is-static"],
    "isScrollable": [4, "is-scrollable"],
    "isCentered": [4, "is-centered"],
    "isFullScreen": [4, "is-full-screen"],
    "fullScreenFrom": [1, "full-screen-from"],
    "modalSize": [1, "modal-size"],
    "showCloseButton": [4, "show-close-button"],
    "footerActionPlacement": [1, "footer-action-placement"]
  }]);
function defineCustomElement$1() {
  if (typeof customElements === "undefined") {
    return;
  }
  const components = ["d-modal", "d-icon"];
  components.forEach(tagName => { switch (tagName) {
    case "d-modal":
      if (!customElements.get(tagName)) {
        customElements.define(tagName, DModal$1);
      }
      break;
    case "d-icon":
      if (!customElements.get(tagName)) {
        defineCustomElement$2();
      }
      break;
  } });
}

const DModal = DModal$1;
const defineCustomElement = defineCustomElement$1;

export { DModal, defineCustomElement };
