// @dynamic-framework/ui/components/d-offcanvas.js
import { proxyCustomElement, HTMLElement, createEvent, h } from '@stencil/core/internal/client';
import { P as PREFIX_BS } from './component-config.js';
import { d as defineCustomElement$2 } from './d-icon2.js';

const DOffcanvas$1 = /*@__PURE__*/ proxyCustomElement(class DOffcanvas extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
    this.eventClose = createEvent(this, "eventClose", 7);
    this.closeHandler = () => {
      this.eventClose.emit();
    };
    this.name = undefined;
    this.isStatic = undefined;
    this.isScrollable = undefined;
    this.showCloseButton = undefined;
    this.openFrom = 'end';
    this.footerActionPlacement = 'fill';
  }
  componentWillLoad() {
    this.header = !!this.el.querySelector('[slot="header"]');
    this.body = !!this.el.querySelector('[slot="body"]');
    this.footer = !!this.el.querySelector('[slot="footer"]');
  }
  render() {
    return (h("div", Object.assign({ class: `offcanvas offcanvas-${this.openFrom} show`, id: this.name, tabindex: "-1", "aria-labelledby": `${this.name}Label`, "aria-hidden": "false" }, this.isStatic && ({
      [`data-${PREFIX_BS}backdrop`]: 'static',
      [`data-${PREFIX_BS}keyboard`]: 'false',
    }), this.isScrollable && ({
      [`data-${PREFIX_BS}scroll`]: 'true',
      [`data-${PREFIX_BS}keyboard`]: 'false',
    })), (this.header || this.showCloseButton) && (h("div", { class: "offcanvas-header" }, this.header && (h("div", { class: "d-offcanvas-slot" }, h("slot", { name: "header" }))), this.showCloseButton && (h("button", { type: "button", class: "d-offcanvas-close", "aria-label": "Close", onClick: this.closeHandler }, h("d-icon", { icon: "x-lg" }))))), this.body && (h("div", { class: "d-offcanvas-slot offcanvas-body" }, h("slot", { name: "body" }))), this.footer && (h("div", { class: `d-offcanvas-slot d-offcanvas-footer d-offcanvas-action-${this.footerActionPlacement}` }, h("slot", { name: "footer" })))));
  }
  get el() { return this; }
}, [4, "d-offcanvas", {
    "name": [1],
    "isStatic": [4, "is-static"],
    "isScrollable": [4, "is-scrollable"],
    "showCloseButton": [4, "show-close-button"],
    "openFrom": [1, "open-from"],
    "footerActionPlacement": [1, "footer-action-placement"]
  }]);
function defineCustomElement$1() {
  if (typeof customElements === "undefined") {
    return;
  }
  const components = ["d-offcanvas", "d-icon"];
  components.forEach(tagName => { switch (tagName) {
    case "d-offcanvas":
      if (!customElements.get(tagName)) {
        customElements.define(tagName, DOffcanvas$1);
      }
      break;
    case "d-icon":
      if (!customElements.get(tagName)) {
        defineCustomElement$2();
      }
      break;
  } });
}

const DOffcanvas = DOffcanvas$1;
const defineCustomElement = defineCustomElement$1;

export { DOffcanvas, defineCustomElement };
