// @dynamic-framework/ui/components/d-quick-action-switch.js
import { proxyCustomElement, HTMLElement, createEvent, h } from '@stencil/core/internal/client';
import { d as defineCustomElement$2 } from './d-input-switch2.js';

const DQuickActionSwitch$1 = /*@__PURE__*/ proxyCustomElement(class DQuickActionSwitch extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
    this.eventClick = createEvent(this, "eventClick", 7);
    this.clickHandler = (event) => {
      event.stopPropagation();
      this.eventClick.emit(this.isChecked);
    };
    this.innerId = undefined;
    this.name = undefined;
    this.label = undefined;
    this.hint = undefined;
    this.isChecked = undefined;
    this.isDisabled = undefined;
  }
  render() {
    return (h("button", { class: "d-quick-action-switch", onClick: this.clickHandler }, h("div", { class: "d-quick-action-switch-content" }, h("d-input-switch", { innerId: this.innerId, name: this.name, isDisabled: this.isDisabled, isChecked: this.isChecked, isReadonly: true }), h("label", { class: "d-quick-action-switch-label", htmlFor: this.innerId }, this.label)), h("div", { class: "d-quick-action-switch-hint" }, this.hint)));
  }
}, [0, "d-quick-action-switch", {
    "innerId": [1, "inner-id"],
    "name": [1],
    "label": [1],
    "hint": [1],
    "isChecked": [4, "is-checked"],
    "isDisabled": [4, "is-disabled"]
  }]);
function defineCustomElement$1() {
  if (typeof customElements === "undefined") {
    return;
  }
  const components = ["d-quick-action-switch", "d-input-switch"];
  components.forEach(tagName => { switch (tagName) {
    case "d-quick-action-switch":
      if (!customElements.get(tagName)) {
        customElements.define(tagName, DQuickActionSwitch$1);
      }
      break;
    case "d-input-switch":
      if (!customElements.get(tagName)) {
        defineCustomElement$2();
      }
      break;
  } });
}

const DQuickActionSwitch = DQuickActionSwitch$1;
const defineCustomElement = defineCustomElement$1;

export { DQuickActionSwitch, defineCustomElement };
