// @dynamic-framework/ui/components/d-quick-action-select.js
import { proxyCustomElement, HTMLElement, createEvent, h } from '@stencil/core/internal/client';

const DQuickActionSelect$1 = /*@__PURE__*/ proxyCustomElement(class DQuickActionSelect extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
    this.eventChange = createEvent(this, "eventChange", 7);
    this.changeHandler = (event) => {
      this.eventChange.emit(event.target.value);
    };
    this.innerId = undefined;
    this.name = undefined;
    this.value = undefined;
    this.line1 = undefined;
    this.line2 = undefined;
    this.isSelected = undefined;
  }
  render() {
    return (h("label", { class: "d-quick-action-select", htmlFor: this.innerId }, h("input", { id: this.innerId, type: "radio", name: this.name, value: this.value, checked: this.isSelected, onChange: this.changeHandler }), h("span", { class: "d-quick-action-select-line1" }, this.line1), h("span", { class: "d-quick-action-select-line2" }, this.line2)));
  }
}, [0, "d-quick-action-select", {
    "innerId": [1, "inner-id"],
    "name": [1],
    "value": [1],
    "line1": [1, "line-1"],
    "line2": [1, "line-2"],
    "isSelected": [4, "is-selected"]
  }]);
function defineCustomElement$1() {
  if (typeof customElements === "undefined") {
    return;
  }
  const components = ["d-quick-action-select"];
  components.forEach(tagName => { switch (tagName) {
    case "d-quick-action-select":
      if (!customElements.get(tagName)) {
        customElements.define(tagName, DQuickActionSelect$1);
      }
      break;
  } });
}

const DQuickActionSelect = DQuickActionSelect$1;
const defineCustomElement = defineCustomElement$1;

export { DQuickActionSelect, defineCustomElement };
