// @dynamic-framework/ui/components/d-quick-action-check.js
import { proxyCustomElement, HTMLElement, createEvent, h } from '@stencil/core/internal/client';
import { d as defineCustomElement$2 } from './d-input-check2.js';

const DQuickActionCheck$1 = /*@__PURE__*/ proxyCustomElement(class DQuickActionCheck extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
    this.eventChange = createEvent(this, "eventChange", 7);
    this.changeHandler = (event) => {
      event.stopPropagation();
      this.eventChange.emit(event.target.value);
    };
    this.innerId = undefined;
    this.name = undefined;
    this.value = undefined;
    this.line1 = undefined;
    this.line2 = undefined;
    this.line3 = undefined;
    this.isChecked = undefined;
  }
  render() {
    return (h("label", { class: "d-quick-action-check", htmlFor: this.innerId }, h("d-input-check", { innerId: this.innerId, type: "radio", name: this.name, value: this.value, isChecked: this.isChecked, onEventChange: this.changeHandler }), h("div", { class: "d-quick-action-check-detail" }, h("span", { class: "d-quick-action-check-line1" }, this.line1), h("span", { class: "d-quick-action-check-line2" }, this.line2)), h("span", { class: "d-quick-action-check-line3" }, this.line3)));
  }
}, [0, "d-quick-action-check", {
    "innerId": [1, "inner-id"],
    "name": [1],
    "value": [1],
    "line1": [1, "line-1"],
    "line2": [1, "line-2"],
    "line3": [1, "line-3"],
    "isChecked": [4, "is-checked"]
  }]);
function defineCustomElement$1() {
  if (typeof customElements === "undefined") {
    return;
  }
  const components = ["d-quick-action-check", "d-input-check"];
  components.forEach(tagName => { switch (tagName) {
    case "d-quick-action-check":
      if (!customElements.get(tagName)) {
        customElements.define(tagName, DQuickActionCheck$1);
      }
      break;
    case "d-input-check":
      if (!customElements.get(tagName)) {
        defineCustomElement$2();
      }
      break;
  } });
}

const DQuickActionCheck = DQuickActionCheck$1;
const defineCustomElement = defineCustomElement$1;

export { DQuickActionCheck, defineCustomElement };
